core/templates/services/literally-canvas-helper.service.ts

```diff
@@ -0,0 +1,184 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview LiterallyCanvas editor helper service.
+ */
+
+
+angular.module('oppia').factory('LiterallyCanvasHelperService', [
+  function() {
+    const getPoints = (x, y, angle, width, length) => [
+      {
+        x: x + (Math.cos(angle + Math.PI / 2) * width) / 2,
+        y: y + (Math.sin(angle + Math.PI / 2) * width) / 2,
+      },
+      {
+        x: x + Math.cos(angle) * length,
+        y: y + Math.sin(angle) * length,
+      },
+      {
+        x: x + (Math.cos(angle - Math.PI / 2) * width) / 2,
+        y: y + (Math.sin(angle - Math.PI / 2) * width) / 2,
+      },
+    ];
+    const arrow = {
+      svg(x, y, angle, width, color, position, length = null) {
+        if (length === null) {
+          length = 0;
+        }
+        length = length || width;
+        const points = getPoints(x, y, angle, width, length);
+
+        return ('<polygon id="' + position + '" fill="' + color + '" ' +
+        'stroke="none" points="' + (points.map(function(p) {
+          return p.x + ',' + p.y;
+        }).join(' ')) + '" />');
+      }
+    };
+    return {
+      rectangleSVGRenderer: function(shape) {
```

Comment:
Can you add a short description for each of these funtions?

Reply:
Done.